### PR TITLE
:recycle: Viewport refactor and improvements

### PR DESCRIPTION
**Pull request by Alotor**
Refactor of the viewport and several enhancements to the selection and movement.

### CODE REVIEW

File: frontend/src/app/main/data/workspace/selection.cljs [hunk 1]
```diff
@@ -84,8 +84,10 @@
 
 ;; --- Toggle shape's selection status (selected or deselected)
 
-(defn select-shape
-  ([id] (select-shape id false))
+(defn select-shape-
```
Comment (niwinz) [inline frontend/src/app/main/data/workspace/selection.cljs]: Why the `-` suffix on the function name?

File: common/app/common/pages/helpers.cljc [hunk 2]
```diff
@@ -410,3 +459,29 @@
         [parent-idx _] (d/seek (fn [[idx child-id]] (= child-id shape-id))
                                (d/enumerate (:shapes parent)))]
     parent-idx))
+
+(defn remove-children
```
Comment (niwinz) [inline common/app/common/pages/helpers.cljc]: Looks like this function has the same purpose than `app.common.page.helpers/clean-loops`, i'm wrong?

File: frontend/src/app/main/ui/workspace/viewport/widgets.cljs [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+   [app.main.ui.hooks :as hooks]
+   [app.main.ui.workspace.shapes.outline :refer [outline]]
+   [app.main.ui.workspace.shapes.path.actions :refer [path-actions]]
+   [clojure.set :as set]
+   [rumext.alpha :as mf]
+   [app.common.pages :as cp]
+   [app.util.dom :as dom]
+   ))
+
+(mf/defc coordinates
```
Comment (niwinz) [inline frontend/src/app/main/ui/workspace/viewport/widgets.cljs]: Duplicate of `app.main.ui.workspace.coordinates/coordinates`.

File: frontend/src/app/main/data/workspace/selection.cljs (hunk 1 above)
Comment (Alotor) [reply frontend/src/app/main/data/workspace/selection.cljs]: ups! it's something i did when i was refactoring and i forgot to undo. I'll revert it

File: frontend/src/app/main/ui/workspace/viewport/widgets.cljs (hunk 3 above)
Comment (Alotor) [reply frontend/src/app/main/ui/workspace/viewport/widgets.cljs]: I'll remove the one in widgets.

File: common/app/common/pages/helpers.cljc (hunk 2 above)
Comment (Alotor) [reply common/app/common/pages/helpers.cljc]: You're right, i'll change to that function.

**Pull request merged**